src/utils/bintools.ts

```diff
@@ -284,11 +284,36 @@ export default class BinTools {
   addressToString = (hrp:string, chainid:string, bytes:Buffer)
   :string => `${chainid}-${bech32.encode(hrp, bech32.toWords(bytes))}`;
 
-  stringToAddress = (address:string):Buffer => {
-    const parts:Array<string> = address.trim().split('-');
-    if(parts[1].startsWith("0x") || parts[1].match(/^[0-9A-F]+$/i)){
+  stringToAddress = (address: string, hrp?: string): Buffer => {
+    const parts: string[] = address.trim().split('-');
+
+    if(parts[0].length < 1) {
+      throw new Bech32Error('Error - Valid address must have prefix before -');
+    }
+
+    if(parts.length < 2) {
+      throw new Bech32Error('Error - Valid address should include -');
+    }
+
+    // ETH-style address
+    if(parts[1].startsWith("0x") || parts[1].match(/^[0-9A-F]+$/i)) {
```

Comment:
We should migrate `parts[1].match(/^[0-9A-F]+$/i)` to be:
```
web3.utils.isAddress('0xc1912fee45d61c87cc5ea59dae31190fffff232d');
```

This does:
```
Checks if a given string is a valid Ethereum address. It will also check the checksum, if the address has upper and lowercase letters.
```

Reply:
You sure it's a good idea to bundle web3 w/ avalanchejs?

Reply:
Fixed